test/distributed/_composable/fsdp/test_fully_shard_init.py

```diff
@@ -270,5 +270,81 @@ def test_get_param_module_infos_duplicates(self):
         ]
 
 
+class TestFullyShardShardedParameterTensor(FSDPTestMultiThread):
+    @property
+    def world_size(self) -> int:
+        return 2
+
+    @unittest.skipIf(not TEST_CUDA, "no cuda")
+    def test_shard_tensor_parameters(self):
+        # Use odd dim sizes to test uneven shards
+        model = nn.Sequential(*[MLP(3, dim_multiplier=3) for _ in range(3)])
+        orig_params = [param.detach().clone() for param in model.parameters()]
+        fully_shard(model)
+        sharded_params = list(model.parameters())
+        self._check_1d_sharded_parameters(orig_params, sharded_params)
+
+        model = nn.Sequential(*[MLP(3, dim_multiplier=3) for _ in range(3)])
+        model[0].in_proj = model[1].in_proj
+        orig_params = [param.detach().clone() for param in model.parameters()]
+        fully_shard(model)
+        sharded_params = list(model.parameters())
+        self._check_1d_sharded_parameters(orig_params, sharded_params)
+
+    def _check_1d_sharded_parameters(
+        self, orig_params: List[nn.Parameter], sharded_params: List[nn.Parameter]
+    ):
+        self.assertEqual(len(orig_params), len(sharded_params))
+        global_mesh = init_device_mesh("cuda", (self.world_size,))
+        for orig_param, sharded_param in zip(orig_params, sharded_params):
+            self.assertIsInstance(sharded_param, DTensor)
+            self.assertEqual(sharded_param.device_mesh, global_mesh)
+            self.assertEqual(sharded_param.size(), orig_param.size())
+            self.assertEqual(sharded_param.stride(), orig_param.stride())
+            self.assertEqual(sharded_param._spec.placements, (Shard(0),))
+            chunks = torch.chunk(orig_param, self.world_size, dim=0)
+            self.assertEqual(sharded_param._local_tensor, chunks[self.rank])
```

Comment:
hmmm for uneven sharding, looks like this test should fail?
i.e. `chunks = torch.chunk(orig_param, self.world_size, dim=0)` should be unpadded while the sharded_param._local_tensor should be already padded?

Reply:
`sharded_param._local_tensor` is a view into the padded tensor, which FSDP maintains, so `sharded_param._local_tensor` is not padded.